### PR TITLE
Update www-and-apex-domain-recommendation.md

### PR DESCRIPTION
**Pull request by JOJ0**
### Why:

The current docs were misleading to me and I had to ask github support why it doesn't work as expected. Support agent Tammy Metz was very helpful and clarified that actually only the special subdomain www is redirected to the apex domain.

### What's being changed:

Trying to clarify that the www subdomain is treated differently compared to subdomains with other names.

### CODE REVIEW

File: data/reusables/pages/www-and-apex-domain-recommendation.md [hunk 1]
```diff
@@ -1 +1 @@
...
+If you are using an apex domain as your custom domain, we recommend also setting up a `www` subdomain. If you configure the correct records for each domain type through your DNS provider, {% data variables.product.prodname_pages %} will automatically create redirects between the domains. For example, if you configure `www.example.com` as your custom domain for your site, and you have `ALIAS` (alternatively `ANAME` or `A` records) and `CNAME` records set up for the apex and `www` domains, then `example.com` will redirect to `www.example.com`. Note that this is only the case for the `www` subdomains. Automatic redirects are not created for subdomains like e.g `blog`. For more information, see "[Managing a custom domain for your {% data variables.product.prodname_pages %} site](/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site/#configuring-a-subdomain)."
```
Comment (tmetz) [inline data/reusables/pages/www-and-apex-domain-recommendation.md:1]:
```suggestion
If you are using an apex domain as your custom domain, we recommend also setting up a `www` subdomain. If you configure the correct records for each domain type through your DNS provider, {% data variables.product.prodname_pages %} will automatically create redirects between the domains. For example, if you configure `www.example.com` as your custom domain for your site, and you have `ALIAS` (alternatively `ANAME` or `A` records) and `CNAME` records set up for the apex and `www` domains, then `example.com` will redirect to `www.example.com`. Note that this is only the case for the `www` subdomains. Automatic redirects are not created for subdomains such as, e.g., `blog`. For more information, see "[Managing a custom domain for your {% data variables.product.prodname_pages %} site](/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site/#configuring-a-subdomain)."
```

**Review by tmetz — DISMISSED** [review]
Thanks again for taking the time to submit these PRs @JOJ0!  Again, I'll defer to our docs team, but this looks good to me (just suggesting one tweak).

**Comment by janiceilene** [comment]
@JOJ0 Thanks so much for opening a PR! I'll get this triaged for review 💖

**Review by lucascosti — APPROVED** [review]
Thanks heaps for this too, @JOJ0 and @tmetz! I've pushed an extra commit (https://github.com/github/docs/pull/4956/commits/53b1375a76a64a0df85791103b8d55af18a9527a) to reword it a little, and fix up the link in the resuable (it's shouldn't link within the same article with the article's name).

I merge it in! 🚂

**Comment by lucascosti** [comment]
🤔 One test seems to be stuck, so I'm to close and reopen this PR to unstick it...

File: data/reusables/pages/www-and-apex-domain-recommendation.md (hunk 1 above)
Comment (JOJ0) [reply data/reusables/pages/www-and-apex-domain-recommendation.md]: alright, that's better yes!

**Pull request merged**